Set minHeight in MapWrapper (formerly GLWrapper) and DemoIframe

katydecorah opened the pull request:
_This PR merges into #393._

This PR:
- Renames GLWrapper to MapWrapper to better describe how you can use the component.
- Requires a `height` for MapWrapper to prevent a layout shift while mapbox-supported makes a decision.
- Required a `title` for DemoIframe to improve accessibility.

## How to test

* [ ] Open /MapWrapper in test cases app
* [ ] Open /DemoIframe in test cases app
* [ ] Preview these maps/pages on staging (I created a beta release from this branch and pushed to /studio-manual https://github.com/mapbox/studio-manual/pull/138):
  * [ ] /studio-manual/examples/3d-buildings/  
  * [ ] /studio-manual/guides/

## QA checklist

<!-- complete this checklist when adding a new component or package -->

- [x] No errors logged to console.
- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `#.#.#`.
- [x] Component is accessible at mobile-size viewport.
- [x] Component is accessible at tablet-size viewport.
- [x] Component is accessible at laptop-size viewport.
- [x] Component is accessible at big-monitor-size viewport.
- [x] Create a PR in a site repo, copy the component, and test it. Push to staging and let the reviewer know they can also test the component there.

Open the test cases app locally on:

- [x] Chrome, no errors logged to console.
- [x] Firefox, no errors logged to console.
- [x] Safari, no errors logged to console.
- [x] Edge, no errors logged to console.
- [x] IE11, no errors logged to console.
- [x] Mobile Safari, no errors logged to console.
- [x] Android Chrome, no errors logged to console.

katydecorah commented:
This is ready for review @colleenmcginnis, I updated the original post with some reviewing suggestions.

colleenmcginnis commented on src/components/demo-iframe/demo-iframe.js:
```diff
@@ -1,60 +1,24 @@
 
 DemoIframe.propTypes = {
-  src: PropTypes.string.isRequired, // absolute URL src for iframe
-  gl: PropTypes.bool, // set to false if the iframe does not use GL
```
Does removing the `gl` prop prevent us from using this component with mapbox.js maps? Or we just don't care if an error message is shown for mapbox.js maps because we won't be supporting browsers that don't work with Mapbox GL JS anyway?

katydecorah replied:
Ah, I didn't think of that use case! Maybe we should revert to adding the `gl` prop back.

katydecorah commented:
@colleenmcginnis this is ready for review again. Per chat, we will be pausing the lazy load work for when we're able to dedicate more time to updating Batfish to support this feature. In the meantime, I've reworked the goal of this PR:

1. Improve CLS (cumulative layout shift) for maps by requiring `height` prop.
2. Improve accessibility of iframes by requiring `title` prop.

I updated the original post again (reviewing section is pretty much the same).

You can see the CLS improvement most significantly on /studio-manual/examples/3d-buildings/. Moving one needle at a time!

Before | After
--|--
![image](https://user-images.githubusercontent.com/2180540/104648837-c0c77880-5681-11eb-8f0e-e2a98981fe09.png) | ![image](https://user-images.githubusercontent.com/2180540/104648847-c3c26900-5681-11eb-85f1-087fbca2a34a.png)

colleenmcginnis commented on src/components/demo-iframe/demo-iframe.js:
```diff
@@ -26,35 +22,53 @@ export default class DemoIframe extends React.Component {
+      ? src.replace('MapboxAccessToken', mapboxAccessToken)
+      : src;
+
+    const iframeHeight = 400;
```
Should this be a default prop value instead? Otherwise defining the height like in the [last test case](https://github.com/mapbox/dr-ui/pull/395/files#diff-3ccba7f92bbe8bb3c98f3082aa91935efade014657d4e9860f50dd8127ce7dcdR38 ) won't work, right? Or I guess the alternative would be to remove `height` from the test case and always use a height of 400. I don't have a strong opinion either way!

katydecorah replied:
@colleenmcginnis Oops, didn't mean to add it to that testcase! I'll remove. We had DemoIframes to always be 400px and I haven't heard a reason to change that. Unless you think we should make it variable?

colleenmcginnis replied:
Nope! Let's leave it at 400px. I do love consistency ... including in the heights of my iframes!

colleenmcginnis reviewed (approved):
🖼️
